### PR TITLE
Simplify op heads interface

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/op_heads_store.rs [hunk 1]
```diff
@@ -121,20 +121,25 @@ pub fn resolve_op_heads<E>(
...
+        op_heads_store.update_op_heads(&ancestor_op_heads, op_head.id());
+        return Ok(op_head);
     }
 
     op_heads.sort_by_key(|op| op.store_operation().metadata.end_time.timestamp.clone());
     match resolver(op_heads) {
         Ok(new_op) => {
-            op_heads_store.update_op_heads(new_op.parent_ids(), new_op.id());
+            let mut old_op_heads = ancestor_op_heads;
+            old_op_heads.extend(new_op.parent_ids().clone());
```
Comment (yuja) [inline lib/src/op_heads_store.rs:135]: nit: `extend_from_slice()` can be used

File: lib/src/op_heads_store.rs [hunk 2]
```diff
@@ -121,20 +121,25 @@ pub fn resolve_op_heads<E>(
...
+    let ancestor_op_heads = op_head_ids_before
+        .difference(&op_head_ids_after)
+        .cloned()
+        .collect_vec();
     let mut op_heads = filtered_op_heads.into_iter().collect_vec();
 
     // Return without creating a merge operation
     if op_heads.len() == 1 {
-        return Ok(op_heads.pop().unwrap());
+        let op_head = op_heads.pop().unwrap();
```
Comment (yuja) [inline lib/src/op_heads_store.rs:126]: nit: could be `if let [op_head] = &*op_heads`, but it wouldn't work if `op_heads` were `HashSet`. I don't know which is better.

File: lib/src/simple_op_heads_store.rs [hunk 3]
```diff
@@ -58,18 +58,7 @@ struct SimpleOpHeadsStoreLock {
...
-    }
-
-    fn update_op_heads(&self, old_ids: &[OperationId], new_id: &OperationId) {
-        self.add_op_head(new_id);
-        for old_id in old_ids {
-            self.remove_op_head(old_id)
-        }
-    }
-
+impl SimpleOpHeadsStore {
```
Comment (yuja) [inline lib/src/simple_op_heads_store.rs:61]: nit: maybe better to merge in to the previous `impl` block?

**Review by yuja — APPROVED** [review]
Nice.

File: lib/src/simple_op_heads_store.rs (hunk 3 above)
Comment (martinvonz) [reply lib/src/simple_op_heads_store.rs]: Oops, yes.

File: lib/src/op_heads_store.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/op_heads_store.rs]: Done.

File: lib/src/op_heads_store.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/op_heads_store.rs]: Done.

**Pull request merged**